extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -742,6 +729,27 @@ export class AmpAutocomplete extends AMP.BaseElement {
     );
   }
 
+  /**
+   * Requests remote data source, if provided, on first user interaction.
+   * @return {!Promise}
+   * @private
+   */
+  checkFirstInteractionAndMaybeFetchData_() {
+    if (!this.interacted_ && this.element.hasAttribute('src')) {
+      this.interacted_ = true;
+      const remoteDataPromise = this.getRemoteData_().catch(e => {
+        this.displayFallback_(e);
+      });
+      return remoteDataPromise.then(remoteData => {
+        if (remoteData) {
```

Comment:
This reads a bit awkward, since the catch above returns the promise chain into a non-error state, and you need to check for data here.

This would be cleaner as:

```javascript
return this.getRemoteData_().then(remoteData => {
  this.sourceData_ = remoteData;
  this.filterDataAndRenderResults_(this.sourceData_);
}, e => {
  this.displayFallback_(e);
});
```

When a catch is used as the second argument to `.then`, it does not apply to the `,then` portion (the first function), but only to the original Promise. In both cases above, the returned Promise is returned to a non-failing state as before.